### DISCUSSION TITLE
自动注入的 umi.js 文件应该放到通过 scripts 注入的文件后面

### DISCUSSION THREAD

**Discussion by huarse** [umijs/umi#8343] [General]
如题，项目配置里面 externals 了 react 等依赖包，然后需要通过 scripts 注入到HTML里面，但是发现 umi.js 竟然放到了这些文件前面，这样会导致依赖问题；

PS：为什么之前的 document.ejs 的方式不支持了？是出于什么考虑呢？

**Comment by fz6m** [comment]
目前解决方式可参考 https://github.com/umijs/umi-next/issues/868#issuecomment-1151088426
